test/ibmq/runtime/test_runtime_integration.py

```diff
@@ -154,6 +156,45 @@ def test_upload_program(self):
         self.assertTrue(program)
         self.assertEqual(max_execution_time, program.max_execution_time)
 
+    def test_update_program_bytes(self):
+        """Test updating a program via byte-encoded string.
+        NOTE: When a Qiskit Runtime API endpoint is created to GET
+         a runtime program in plaintext, update this test to verify the
+         program is modified.
+        """
+        # Create some program
+        program_id = self._upload_program()
+        self.assertTrue(program_id)
+        # Update the program data (last 2 lines differ from original)
+        new_program = self.RUNTIME_PROGRAM.replace('warnings.warn("this is a stderr message")',
+                                                   'warnings.warn("this is not a stderr message")')
+        # Execute with bytes
+        self.provider.runtime.update_program(program_id, new_program.encode())
+
+    @skipIf(os.name == 'nt', 'Test not supported on Windows')
```

Comment:
How do we support this for windows?

Reply:
The test can use normal test files instead of using the `tempfile` library. `tempfile` has a bit of [an issue ](https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile)with windows NT. It's a small change changing to normal files, it just would look a bit less concise because it would need try/catch to ensure temp files are closed and deleted.

Reply:
Sounds good. Could we move this logic to a private function in this file itself so that it can be reused for update_program as well? We can change the update_program logic in a separate PR.

Reply:
Yes let's use normal test files. Thanks.

Reply:
[Updated!](https://github.com/Qiskit/qiskit-ibmq-provider/pull/1008/files#diff-b2e8d26399a76d1abcdbaefe0f2629ea83c6c8fe363347e769c8400c5599c0b6R335-R339) Uses normal test files.